tests/components/plugwise/test_init.py

```diff
@@ -225,6 +233,38 @@ async def test_migrate_unique_id_relay(
     assert entity_migrated.unique_id == new_unique_id
 
 
+async def test_entry_migration(
+    hass: HomeAssistant, mock_smile_anna_2: MagicMock, snapshot: SnapshotAssertion
+) -> None:
+    """Test config entry version 1 -> 2 migration."""
+    entry = MockConfigEntry(
+        title="My Plugwise",
+        domain=DOMAIN,
+        data={
+            CONF_HOST: "127.0.0.1",
+            CONF_MAC: "AA:BB:CC:DD:EE:FF",
+            CONF_PASSWORD: "test-password",
+            CONF_PORT: 80,
+            CONF_USERNAME: "smile",
+        },
+        minor_version=1,
+        version=1,
+        unique_id="smile98765",
+    )
+
+    entry.runtime_data = MagicMock(api=mock_smile_anna_2)
+    entry.add_to_hass(hass)
+
+    assert entry.version == 1
+    assert entry.minor_version == 1
+
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    # Assert that the migrated entry matches the expected structure
+    assert hass.config_entries.async_get_entry(entry.entry_id) == snapshot
```

Comment:
I do like snapshots, but in this case I would rather see a test that tests if the entry is changed to what we expect. Snapshot tests are nice to fixate data objects, but they don't tell us the meaning behind the change nor what we expect it to test

Reply:
OK, I took inspiration from a wrong integration then 🥲

Reply:
Changed as requested.